src/main/kotlin/com/hj/leetcode/kotlin/problem2492/Solution2.kt

```diff
@@ -0,0 +1,57 @@
+package com.hj.leetcode.kotlin.problem2492
+
+/**
+ * LeetCode page: [2492. Minimum Score of a Path Between Two Cities](https://leetcode.com/problems/minimum-score-of-a-path-between-two-cities/);
+ */
+class Solution2 {
+    /* Complexity:
+     * Time O(n+E) and Space O(n) where E is the size of roads;
+     */
+    fun minScore(n: Int, roads: Array<IntArray>): Int {
+        val unionFind = buildUnionFindOfCity(n, roads)
+        var minScore = Int.MAX_VALUE
+        for ((u, _, distance) in roads) {
+            if (unionFind.inSameUnion(u, 1) && distance < minScore) {
+                minScore = distance
+            }
+        }
```

Comment:
I suggest introducing a new function that uses the unionFind and roads to find the minScore